pysmt/solvers/qelim.py

```diff
@@ -42,3 +48,39 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         the associated resources.
         """
         del self
+
+
+class ShannonQuantifierEliminator(QuantifierEliminator, IdentityDagWalker):
+
+    LOGICS = [pysmt.logics.BOOL]
+
+    def __init__(self, environment, logic=None):
+        IdentityDagWalker.__init__(self, env=environment)
+        self.logic = logic
+
+    def eliminate_quantifiers(self, formula):
+        logic = get_logic(formula, self.env)
+        if not logic <= pysmt.logics.BOOL:
+            raise NotImplementedError("Shannon quantifier elimination only "\
```

Comment:
I see. What do you think of this:

- This QE declares support only for BOOL
- The check within the QE is done only on quantified variables

This solves the issue of automatic selection (it never gets selected incorrectly), but one can force the use of this qe ```qelim(f, name="shannon")```, that will raise an error if quantified variables are not bool.